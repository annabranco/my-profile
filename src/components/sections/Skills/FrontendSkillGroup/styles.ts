import styled, { css } from 'styled-components';

interface FrontendGroupProps extends React.ComponentPropsWithoutRef<'div'> {
  visible?: boolean;
}

export const FrontendGroup = styled.div<FrontendGroupProps>`
  grid-column: span 4;
  margin-top: 100%;
  opacity: 0;
  transition: margin 3s ease, opacity 3s ease;

  ${({ visible }) =>
    visible &&
    css`
      margin-top: 0;
      opacity: 1;
    `}
`;
FrontendGroup.displayName = 'Frontend Group';

interface LogoProps extends React.ComponentPropsWithoutRef<'img'> {
  name?: string;
}

export const Logo = styled.img<LogoProps>`
  height: 120%;

  &.icon--react {
    height: 30px;
  }
  ${({ name }) =>
    name === 'React' &&
    css`
      height: 130%;
      transform: translate(0, -2px);
    `}
  @media all and (min-width: 768px) {
    height: 100%;
    font-size: 3.5rem;

    ${({ name }) =>
      name === 'React' &&
      css`
        height: 120%;
        transform: translate(0, -2px);
      `}
  }
`;
Logo.displayName = 'Logo';

export const LogoWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 30px;
  width: 60px;

  @media all and (min-width: 768px) {
    height: 68px;
  }
`;
LogoWrapper.displayName = 'Logo Wrapper';

export const SkillsWrapper = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-around;
  flex-wrap: wrap;
`;
SkillsWrapper.displayName = 'Skills Wrapper';
