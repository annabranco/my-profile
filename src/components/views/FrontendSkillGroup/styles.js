import styled, { css } from 'styled-components';

export const FrontendGroup = styled.div`
  grid-column: span 4;
  margin-top: 100%;
  opacity: 0;
  transition: margin 3s ease, opacity 3s ease;

  ${({ visible }) =>
    visible &&
    css`
      margin-top: 0;
      opacity: 1;
    `}
`;
FrontendGroup.displayName = '--- Frontend Group ---';

export const SkillsWrapper = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-around;
`;
SkillsWrapper.displayName = '--- Skills Wrapper ---';

export const LogoWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 68px;
  width: 60px;
`;
LogoWrapper.displayName = '--- Logo Wrapper ---';

export const Logo = styled.img`
  height: 100%;
  font-size: 3.5rem;

  &.icon--react {
    height: 60px;
  }
`;
Logo.displayName = '--- Logo ---';
