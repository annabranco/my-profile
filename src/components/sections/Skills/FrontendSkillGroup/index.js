import React from 'react';
import { bool } from 'prop-types';
import SkillLevel from '../SkillLevel';
import { FrontendGroup, Logo, LogoWrapper, SkillsWrapper } from './styles';
import { Name, SkillItem, Title } from '../styles';
import { skillGroupsPropType } from '../../../../types';
import { isDesktop } from '../../../../utils/device';

const FrontendSkillGroup = ({ details, visible }) => (
  <FrontendGroup visible={visible}>
    <Title group={details.name}>
      <h3>{details.name}</h3>
    </Title>
    <SkillsWrapper>
      {details.skills.map(skill => {
        if (!isDesktop && skill.extended) {
          return null;
        }
        return (
          <SkillItem key={`${skill.order}-${skill.skill}`}>
            <Name>{skill.skill}</Name>

            {skill.logo && (
              <LogoWrapper>
                <Logo src={skill.logo} alt={skill.skill} name={skill.skill} />
              </LogoWrapper>
            )}

            <SkillLevel level={skill.level} skill={skill.skill} />
          </SkillItem>
        );
      })}
    </SkillsWrapper>
  </FrontendGroup>
);

FrontendSkillGroup.propTypes = {
  details: skillGroupsPropType.isRequired,
  visible: bool.isRequired
};

export default FrontendSkillGroup;
