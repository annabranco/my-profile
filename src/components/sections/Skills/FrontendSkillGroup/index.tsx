import { ReactElement } from 'react';
import { bool } from 'prop-types';
import { ISkillGroups } from '../../../../types/interfaces';
import SkillLevel from '../SkillLevel';
import { isDesktop } from '../../../../utils/device';
import { skillGroupsPropType } from '../../../../types/propTypes';
import { FrontendGroup, Logo, LogoWrapper, SkillsWrapper } from './styles';
import { Name, SkillItem, Title } from '../styles';

interface Props {
  details: ISkillGroups;
  visible: boolean;
}

const FrontendSkillGroup = ({ details, visible }: Props): ReactElement => (
  <FrontendGroup visible={visible}>
    <Title group={details.name}>
      <h3>{details.name}</h3>
    </Title>
    <SkillsWrapper>
      {details.skills.map(skill => {
        if (!isDesktop && skill.extended) {
          return null;
        }
        return (
          <SkillItem group={details.name} key={`${skill.order}-${skill.skill}`}>
            <Name>{skill.skill}</Name>

            {skill.logo && (
              <LogoWrapper>
                <Logo src={skill.logo} alt={skill.skill} name={skill.skill} />
              </LogoWrapper>
            )}

            <SkillLevel level={skill.level} skill={skill.skill} />
          </SkillItem>
        );
      })}
    </SkillsWrapper>
  </FrontendGroup>
);

FrontendSkillGroup.propTypes = {
  details: skillGroupsPropType.isRequired,
  visible: bool.isRequired
};

export default FrontendSkillGroup;
