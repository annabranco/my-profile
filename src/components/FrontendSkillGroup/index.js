import React from 'react';
import { bool } from 'prop-types';
import SkillLevel from '../SkillLevel';
import { FrontendGroup, Logo, LogoWrapper, SkillsWrapper } from './styles';
import { Name, SkillItem, Title } from '../Skills/styles';
import { skillGroupsPropType } from '../../types';

const FrontendSkillGroup = ({ details, visible }) => (
  <FrontendGroup visible={visible}>
    <Title group={details.name}>{details.name}</Title>
    <SkillsWrapper>
      {details.skills.map(skill => (
        <SkillItem key={`${skill.order}-${skill.skill}`}>
          <Name>{skill.skill}</Name>

          {skill.logo && (
            <LogoWrapper>
              <Logo src={skill.logo} alt="" />
            </LogoWrapper>
          )}

          <SkillLevel level={skill.level} />
        </SkillItem>
      ))}
    </SkillsWrapper>
  </FrontendGroup>
);

FrontendSkillGroup.propTypes = {
  details: skillGroupsPropType.isRequired,
  visible: bool.isRequired
};

export default FrontendSkillGroup;
