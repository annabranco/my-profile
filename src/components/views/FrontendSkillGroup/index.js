import React from 'react';
import { PropTypes } from 'prop-types';
import SkillLevel from '../SkillLevel';
import { SkillGroupTitle, SkillWrapper, SkillName } from '../Skills/styles';
import { FrontendSkillWrapper, FrontendSkillsWrapper } from './styles';

const FrontendSkillGroup = ({ details }) => (
  <FrontendSkillWrapper>
    <SkillGroupTitle group={details.name}>{details.name}</SkillGroupTitle>
    <FrontendSkillsWrapper>
      {details.skills.map(skill => (
        <SkillWrapper key={skill.skill}>
          <SkillName>{skill.skill}</SkillName>

          {skill.logo && (
            <div className="skills__inner--icon">
              <img src={skill.logo} alt="" className="skills__icon" />
            </div>
          )}

          <SkillLevel level={skill.level} />
        </SkillWrapper>
      ))}
    </FrontendSkillsWrapper>
  </FrontendSkillWrapper>
);

// const get = () => import('../../../images/skills/js.png');

// const FrontendSkillGroup = ({ details }) => {
//   // .then(module => module.default)
//   // .then(path => path);
//   // console.log('$$$ temp2', temp2);
//   // return temp2;

//   return get().then(a => (
//     <FrontendSkillWrapper>
//       <img src={a.default} alt="" />
//       <SkillGroupTitle group={details.name}>{details.name}</SkillGroupTitle>
//       <FrontendSkillsWrapper>
//         {details.skills.map(skill => (
//           <SkillWrapper key={skill.skill}>
//             <SkillName>{skill.skill}</SkillName>
//             {/*
//                   <div className="skills__inner--icon">
//                      <img src={Css} alt="" className="skills__icon" />
//                   </div>
//                   */}
//             <SkillLevel level={skill.level} />
//           </SkillWrapper>
//         ))}
//       </FrontendSkillsWrapper>
//     </FrontendSkillWrapper>
//   ));
// };

FrontendSkillGroup.propTypes = {
  details: PropTypes.string.isRequired
};

export default FrontendSkillGroup;
