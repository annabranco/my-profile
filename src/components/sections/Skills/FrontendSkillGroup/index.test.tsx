import { shallow, ShallowWrapper } from 'enzyme';
import 'jest-styled-components';
import FrontendSkillGroup from '.';
import { getMockState, setupMockProvider } from '../../../../../testing/mocks';

// TODO [10-May -21]: Implement defined testing suite (Anna Branco)

describe('< FrontendSkillGroup >', () => {
  const mockSkillObject = getMockState({
    language: 'en'
  }).skills[0];
  let MockProvider;
  let wrapper: ShallowWrapper;

  beforeAll(() => {
    MockProvider = setupMockProvider({ language: 'en' }).MockProvider;
    wrapper = shallow(
      <MockProvider>
        <FrontendSkillGroup details={mockSkillObject} visible />
      </MockProvider>
    );
  });

  it('should mount', () => {
    expect(wrapper).toBeDefined();
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  xit('should display all skills received by props', () => {});

  xit('should display a logo for all skill', () => {});

  xit('should render on each skill a SkillLevel subcomponent passing a skill level as props', () => {});
});
