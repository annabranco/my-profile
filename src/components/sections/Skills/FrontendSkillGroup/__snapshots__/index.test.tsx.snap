// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< FrontendSkillGroup > matches the snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <FrontendSkillGroup
    details={
      Object {
        "globalLevel": 3,
        "logo": "",
        "name": "Frontend",
        "order": 1,
        "skills": Array [
          Object {
            "level": 3,
            "logo": "https://raw.githubusercontent.com/annabranco/my-profile/master/src/assets/images/skills/js.png",
            "order": 1,
            "skill": "JavaScript",
          },
          Object {
            "level": 3,
            "logo": "https://raw.githubusercontent.com/annabranco/my-profile/master/src/assets/images/skills/react.png",
            "order": 2,
            "skill": "React",
          },
          Object {
            "level": 2,
            "logo": "https://raw.githubusercontent.com/annabranco/my-profile/master/src/assets/images/skills/redux.png",
            "order": 3,
            "skill": "Redux",
          },
          Object {
            "level": 3,
            "logo": "https://raw.githubusercontent.com/annabranco/my-profile/master/src/assets/images/skills/typescript.png",
            "order": 4,
            "skill": "TypeScript",
          },
          Object {
            "level": 3,
            "logo": "https://raw.githubusercontent.com/annabranco/my-profile/master/src/assets/images/skills/frontity.png",
            "order": 5,
            "skill": "Frontity",
          },
          Object {
            "level": 1,
            "logo": "https://raw.githubusercontent.com/annabranco/my-profile/master/src/assets/images/skills/angular.png",
            "order": 6,
            "skill": "Angular",
          },
          Object {
            "extended": true,
            "level": 3,
            "logo": "https://raw.githubusercontent.com/annabranco/my-profile/master/src/assets/images/skills/html5.png",
            "order": 7,
            "skill": "HTML5",
          },
          Object {
            "level": 3,
            "logo": "https://raw.githubusercontent.com/annabranco/my-profile/master/src/assets/images/skills/css3.png",
            "order": 8,
            "skill": "CSS3",
          },
          Object {
            "level": 3,
            "logo": "https://raw.githubusercontent.com/annabranco/my-profile/master/src/assets/images/skills/styled-components.png",
            "order": 9,
            "skill": "Styled",
          },
        ],
      }
    }
    visible={true}
  />
</Provider>
`;
